xen/arch/arm/tee/ffa_msg.c

```diff
@@ -78,3 +87,48 @@ void ffa_handle_msg_send_direct_req(struct cpu_user_regs *regs, uint32_t fid)
                  resp.a4 & mask, resp.a5 & mask, resp.a6 & mask,
                  resp.a7 & mask);
 }
+
+int32_t ffa_handle_msg_send2(struct cpu_user_regs *regs)
+{
+    struct domain *src_d = current->domain;
+    struct ffa_ctx *src_ctx = src_d->arch.tee;
+    const struct ffa_part_msg_rxtx *src_msg;
+    int32_t ret;
+
+    if ( !ffa_fw_support_fid(FFA_MSG_SEND2) )
+    {
+        ret = FFA_RET_NOT_SUPPORTED;
+        goto out;
+    }
+
+    if ( !spin_trylock(&src_ctx->tx_lock) )
+    {
+        ret = FFA_RET_BUSY;
+        goto out;
+    }
+
```

Comment:
w1 Bit[31:16] is MBZ unless at the Non-secure physical instance. Do we want to check that?

Reply:
This will be relaxed in future spec version as SBZ (should be zero) and as we do not use the value i am just ignoring it.